tree/tree.go

```diff
@@ -227,6 +227,24 @@ func (t *Tree) FromRoot(root []byte) (*Tree, error) {
 	}, nil
 }
 
+func (t *Tree) SetRoot(wTx db.WriteTx, root []byte) error {
+	givenTx := true
+	if wTx == nil {
+		givenTx = false
```

Comment:
nit: easier if you do

```
givenTx := wTx != nil
if !givenTx {
```

this way it also mirrors the other `if !givenTx` check, so it's easier to understand they are matching each other